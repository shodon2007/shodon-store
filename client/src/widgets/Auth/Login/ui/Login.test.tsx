import { screen } from '@testing-library/react';
import renderTestApp from 'src/app/tests/helpers/renderTestApp';
import '@testing-library/jest-dom/jest-globals';

describe('jest testing Login.tet.tsx', () => {
	test('testing elements in document', () => {
		renderTestApp('/auth');

		const loginPage = screen.getByTestId('login-page');
		const loginButton = screen.getByTestId('login-button');
		const loginInputs = screen.getAllByTestId('login-input');

		expect(loginPage).toBeInTheDocument();
		expect(loginButton).toBeInTheDocument();
		expect(loginInputs.length).toBe(2);
	});

	test('testing snapshopts', () => {
		renderTestApp('/auth');

		const loginPage = screen.getByTestId('login-page');

		expect(loginPage).toMatchSnapshot();
	});
});
