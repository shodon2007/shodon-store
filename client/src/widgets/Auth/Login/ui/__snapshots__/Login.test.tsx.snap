// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest testing Login.tet.tsx testing snapshopts 1`] = `
<div
  class="inputs"
  data-testid="login-page"
>
  <input
    class="Input "
    data-testid="login-input"
    placeholder="login"
    type="text"
    value=""
  />
  <input
    class="Input "
    data-testid="login-input"
    minlength="8"
    placeholder="password"
    required=""
    type="password"
    value=""
  />
</div>
`;
